lib/chef/cookbook_version.rb

```diff
@@ -96,7 +96,13 @@ def root_dir
     # This is the one and only method that knows how cookbook files'
     # checksums are generated.
     def self.checksum_cookbook_file(filepath)
-      Chef::Digester.generate_md5_checksum_for_file(filepath)
+      if Chef::Config.fips_mode
+        # This will require a chef server that can handle
+        # sha256 checksums
```

Comment:
@danielsdeleo if you ask openssl to take the md5 of something, and fips mode is enabled, it will raise an error. If you ask ruby through `digest/md5`, it will abort the process. This is why we need to use `OpenSSL::Digest` instead of `Digest`. 

Right now, the plan is that we will have different omnibus packages with fips mode enabled. So while a regular build of chef would be able to accept both, the same would not be true for the build with openssl running in fips mode